From 8fa928f1617aba65f45b00f0dcb109f077b7741e Mon Sep 17 00:00:00 2001
From: Alexei Podtelezhnikov <apodtele@gmail.com>
Date: Wed, 13 May 2026 21:31:32 -0400
Subject: [PATCH] * src/type1/t1load.c (T1_Get_Var_Design): Updated.

---
 src/type1/t1load.c | 11 +++++------
 1 file changed, 5 insertions(+), 6 deletions(-)

diff --git a/src/type1/t1load.c b/src/type1/t1load.c
index 1d095f8ac..2ed327ba2 100644
--- a/src/type1/t1load.c
+++ b/src/type1/t1load.c
@@ -677,7 +677,7 @@
     PS_Blend  blend  = t1face->blend;
 
     FT_Fixed  axiscoords[4];
-    FT_UInt   i, nc;
+    FT_UInt   i;
 
 
     if ( !blend )
@@ -687,19 +687,18 @@
                       axiscoords,
                       blend->num_axis );
 
-    nc = num_coords;
     if ( num_coords > blend->num_axis )
     {
       FT_TRACE2(( "T1_Get_Var_Design:"
                   " only using first %u of %u coordinates\n",
                   blend->num_axis, num_coords ));
-      nc = blend->num_axis;
+
+      FT_ARRAY_ZERO( coords + blend->num_axis, num_coords - blend->num_axis );
+      num_coords = blend->num_axis;
     }
 
-    for ( i = 0; i < nc; i++ )
+    for ( i = 0; i < num_coords; i++ )
       coords[i] = mm_axis_unmap( &blend->design_map[i], axiscoords[i] );
-    for ( ; i < num_coords; i++ )
-      coords[i] = 0;
 
     return FT_Err_Ok;
   }
